src/lambda/commands/deploySamApplication.ts

```diff
@@ -300,10 +303,13 @@ function getDefaultWindowFunctions(): WindowFunctions {
     }
 }
 
-function getDefaultSamDeployWizardResponseProvider(regionProvider: RegionProvider): SamDeployWizardResponseProvider {
+function getDefaultSamDeployWizardResponseProvider(
+    regionProvider: RegionProvider,
+    context: Pick<vscode.ExtensionContext, 'asAbsolutePath' | 'globalState'>
+): SamDeployWizardResponseProvider {
     return {
         getSamDeployWizardResponse: async (): Promise<SamDeployWizardResponse | undefined> => {
-            const wizard = new SamDeployWizard(regionProvider)
+            const wizard = new SamDeployWizard(regionProvider, new DefaultSamDeployWizardContext(context))
```

Comment:
Will this function ever be called from unit tests? If so, we can't directly instantiate `DefaultSamDeployWizardContext`.

Reply:
I don't think it would be. This function is already the stub for function `deploySamApplication`

Reply:
We don't have any tests for `DefaultSamDeployWizardContext` or `new SamDeployWizard`. I think this pattern only exists for the tests?: https://github.com/aws/aws-toolkit-vscode/pull/584/files/796bc24bc5449673a79e2fc22800bb8f1b497295#diff-af229da7cae11aa77224c3d2befbb53aR131

I'm not a huge fan of it myself...if you insist, I can attempt a refactor.